usr/lib/linuxmuster-webui/plugins/lmn_linbo_sync/api.py

```diff
@@ -183,19 +184,51 @@ def test_online(host):
 
     :param host: Hostname
     :type host: string
-    :return: OS type
+    :return: OS type (Off, Linbo, OS Linux, OS Windows, OS Unknown)
     :rtype: string
     """
-
-    command=['nmap', '-p', '22', '-O', host]
+    command=["nmap", "-p", "2222,22,135", host, "-oX", "-"]
     r = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=False).stdout.read()
-    if b'Too many fingerprints' in r:
-        return "Linbo"
-    elif b'Linux' in r:
+    xmlRoot = ET.fromstring(r)
+    openPorts = []
+
+    try:
+        hostElement = xmlRoot.findall("host")[0]
+        portsElement = hostElement.findall("ports")[0]
+        scannedPorts = portsElement.findall("port")
+    except:
+        return get_os_from_open_ports([])
```

Comment:
Just return "Off", no need to launch another function;

Reply:
I thought it is cleaner like this, because in case the string `Off` would be changed, it would only have to be changed in one place.

Reply:
Yes, it's cleaner, and I was thinking twice before add this coment, just let it as it is ;)

Reply:
Actually, you are right. No open ports does not mean that the host is down. I will clean this up.

Reply:
I found a cleaner solution. Over here it now returns "Off", but get_os_from_ports cannot return Off anymore.